src/sentry/migrations/0661_create_new_broken_monitor_detection_table.py

```diff
@@ -0,0 +1,51 @@
+# Generated by Django 5.0.2 on 2024-03-04 22:40
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+import sentry.db.models.fields.bounded
+import sentry.db.models.fields.foreignkey
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
+
+    dependencies = [
+        ("sentry", "0660_fix_cron_monitor_invalid_orgs"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="MonitorEnvBrokenDetection",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                ("detection_timestamp", models.DateTimeField()),
+                ("user_notified_timestamp", models.DateTimeField(null=True)),
+                (
+                    "monitor_incident",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE, to="sentry.monitorincident"
+                    ),
+                ),
```

Comment:
With the upcoming crons postgres split you'll need to update `router.py` in getsentry **before** merging this change to ensure that your migration is routed to the correct cluster when it runs.

Reply:
ahh yes good catch, would it just look like this? https://github.com/getsentry/getsentry/pull/13152

Reply:
Precisely that :clap: